File: backend/src/main/java/edu/miu/cs/mtc/api/model/search/CandidateSearch.java
package edu.miu.cs.mtc.api.model.search;

import edu.miu.cs.mtc.api.model.embedded.Skill;
import edu.miu.cs.mtc.api.model.enums.CandidateProfileStatus;
import java.util.List;
import lombok.Data;

@Data
public class CandidateSearch {
  private String id;
  private String firstName;
  private String lastName;
  private String jobTitle;
  private String profilePictureUrl;
  private String summary;
  private List<Skill> skills;
  private List<String> primaryTechnologies;
  private String resumeUrl;
  private CandidateProfileStatus profileStatus;
  private String userId;
  private String profileId;
}

File: frontend/src/models/CandidateSearch.ts
import { CandidateProfileStatus } from './CandidateProfileStatus';
import Skill from './Skill';

export interface CandidateSearch {
  id: string;
  firstName: string;
  lastName: string;
  jobTitle: string;
  profilePictureUrl?: string;
  summary: string;
  skills?: Skill[];
  primaryTechnologies?: string[];
  resumeUrl?: string;
  profileStatus: CandidateProfileStatus;
  userId: string;
  profileId: string;
}
